### PR TITLE
Alert Setting Support

### PR DESCRIPTION
**Pull request by Bhavneet-Sharma**
# Description
Added Support for the Alert settings

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [X] I have performed Ansible Sanity test using --docker default
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Functional Test
- [X] Unit Test

### CODE REVIEW

File: plugins/modules/alert_settings.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            description: End time of CELOG maintenance mode, as a UNIX
+                         timestamp in seconds.
+                         Value 0 indicates that maintenance
+                         mode is still enabled.
+            type: int
+          start:
+            description: Start time of CELOG maintenance mode, as a UNIX
+                         timestamp in seconds.
+            type: int
+          network_pools:
```
Comment (meenakshidembi691) [inline plugins/modules/alert_settings.py:102]: Do we have network pools also in response?

File: plugins/modules/alert_settings.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+- name: Get alert setiings
+  dellemc.powerscale.alert_settings:
+    onefs_host: "{{ onefs_host }}"
+    port_no: "{{ port_no }}"
+    api_user: "{{ api_user }}"
+    api_password: "{{ api_password }}"
+    verify_ssl: "{{ verify_ssl }}"
+
+- name: Enable CELOG maintenance mode
+  dellemc.powerscale.support_assist:
```
Comment (meenakshidembi691) [inline plugins/modules/alert_settings.py:57]: Please change the file name from support_assist to alert_settings

File: plugins/modules/alert_settings.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    onefs_host: "{{ onefs_host }}"
+    port_no: "{{ port_no }}"
+    api_user: "{{ api_user }}"
+    api_password: "{{ api_password }}"
+    verify_ssl: "{{ verify_ssl }}"
+    enable_celog_maintenance_mode: true
+    prune: 1
+
+- name: Disable CELOG maintenance mode
+  dellemc.powerscale.support_assist:
```
Comment (meenakshidembi691) [inline plugins/modules/alert_settings.py:67]: Please change the file name from support_assist to alert_settings

File: tests/unit/plugins/modules/test_alert_settings.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+class TestAlertSettings(PowerScaleUnitBase):
+    alert_args = MockAlertSettingsApi.ALERT_COMMON_ARGS
+
+    @pytest.fixture
+    def module_object(self):
+        """
+        Returns an instance of the `AlertSettings` class for testing purposes.
+
+        :return: An instance of the `AlertSettings` class.
+        :rtype: `SMB`
```
Comment (meenakshidembi691) [inline tests/unit/plugins/modules/test_alert_settings.py:34]: Can we change rtype value

**Comment by meenakshidembi691** [comment]
.tar file is also checkin please check if that is required

File: plugins/modules/alert_settings.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    - Removes all maintenance mode history that is greater than set number of days.
+    - Range of I(prune) is 0 to 4294967295.
+    type: int
+
+notes:
+- The I(check_mode) and idempotency is supported.
+'''
+
+EXAMPLES = r'''
+- name: Get alert setiings
```
Comment (sachin-apa) [inline plugins/modules/alert_settings.py:48]: Are we giving an option to get alert settings from the configuration module ? 
If yes then we need to remove this and it should be part of info module only.

File: playbooks/modules/alert_settings.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  hosts: localhost
+  connection: local
+  vars:
+    onefs_host: "x.x.x.x"
+    api_user: "user"
+    api_password: "xxxx"
+    verify_ssl: false
+
+  tasks:
+    - name: Get alert settings
```
Comment (sachin-apa) [inline playbooks/modules/alert_settings.yml:12]: This should be possible only from a info module.

File: plugins/modules/alert_settings.py (hunk 1 above)
Comment (Bhavneet-Sharma) [reply plugins/modules/alert_settings.py]: Done.

File: plugins/modules/alert_settings.py (hunk 2 above)
Comment (Bhavneet-Sharma) [reply plugins/modules/alert_settings.py]: Done.

File: plugins/modules/alert_settings.py (hunk 3 above)
Comment (Bhavneet-Sharma) [reply plugins/modules/alert_settings.py]: Done.

File: tests/unit/plugins/modules/test_alert_settings.py (hunk 4 above)
Comment (Bhavneet-Sharma) [reply tests/unit/plugins/modules/test_alert_settings.py]: Done.

File: dellemc-powerscale-3.2.0.tar.gz
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply dellemc-powerscale-3.2.0.tar.gz:1]: Done.

File: plugins/modules/alert_settings.py (hunk 5 above)
Comment (Bhavneet-Sharma) [reply plugins/modules/alert_settings.py]: Done.

File: playbooks/modules/alert_settings.yml (hunk 6 above)
Comment (Bhavneet-Sharma) [reply playbooks/modules/alert_settings.yml]: Done.

**Review by felixs88 — DISMISSED** [review]

**Review by felixs88 — APPROVED** [review]

File: docs/modules/alert_settings.rst [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+Managing alert settings on a PowerScale system includes retrieving details of alert settings and enabling or disabling the alert settings.
+
+
+
+Requirements
+------------
+The below requirements are needed on the host that executes this module.
+
+- A Dell PowerScale Storage system.
+- Ansible-core 2.14 or later.
```
Comment (trisha-dell) [inline docs/modules/alert_settings.rst:24]: 2.15

File: docs/modules/alert_settings.rst [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+- Python 3.9, 3.10 or 3.11.
```
Comment (trisha-dell) [inline docs/modules/alert_settings.rst:25]: 3.10, 3.11, and 3.12

File: docs/modules/alert_settings.rst (hunk 7 above)
Comment (Bhavneet-Sharma) [reply docs/modules/alert_settings.rst:24]: Can we fix this with Doc PR

File: docs/modules/alert_settings.rst (hunk 8 above)
Comment (Bhavneet-Sharma) [reply docs/modules/alert_settings.rst:25]: Can we fix this with Doc PR

**Review by Kritika-Bhateja-03 — APPROVED** [review]

**Review by meenakshidembi691 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Review by trisha-dell — APPROVED** [review]

**Pull request merged**